### DISCUSSION TITLE
I cannot delete an image from a public storage bucket.

### DISCUSSION THREAD

**Discussion by michaelpumo** [supabase/supabase#10714] [Questions] [answered]
I have a storage bucket called "images" and when a user uploads an image, it will be placed inside a folder created from their "user_id". So all their images are inside of this folder unique to them.

I can upload just fine but cannot delete the actual files. If I make the request to delete/remove, it does not error in the console but the file still remains in the folder.

```javascript
const { data, error } = await client.storage
  .from('images')
  .remove([`${props.userId}/${props.fileName}`])
```

The remove string is: `26e2d48e-e26e-4f29-a0be-588646f154d8/image-d9758ddb-b32b-4f78-ab21-1891c0445508.jpg`

Here are some screenshots of my setup and RLS policy. I cannot see what I'm missing here.

<img width="789" alt="Screenshot 2022-12-03 at 22 37 38" src="https://user-images.githubusercontent.com/4269460/205465145-b6d0f6c0-8c95-452a-8eda-06f8da5868f9.png">
<img width="914" alt="Screenshot 2022-12-03 at 22 38 33" src="https://user-images.githubusercontent.com/4269460/205465146-228d5d22-098c-429c-8b0c-28c51a2fd670.png">
<img width="1176" alt="Screenshot 2022-12-03 at 22 38 51" src="https://user-images.githubusercontent.com/4269460/205465148-ec374708-a5ed-4604-b3e5-011f1f806f9b.png">

Any ideas what I'm doing wrong here?

**Comment by GaryAustin1** [comment, marked as answer]
Delete requires you also meet select policy.  I don't see a select policy.
This is still required even if a public bucket.  All public does is allow a URL that does not depend on RLS.

**Reply by michaelpumo** [reply]
Thank you. That seems to have worked. I was following the guide here and it says that there's no need for a select if the bucket is public (which it is) so I was a bit confused. https://supabase.com/docs/guides/storage#access-control

**Comment by GaryAustin1** [comment]
`This calls the /object/public/ API endpoint internally. While no authorization is required for accessing objects in a public bucket, proper [access control](https://supabase.com/docs/guides/storage#access-control) is required for other operations like uploading, deleting objects from public buckets as well.`

Mentions delete needing RLS and Postgres delete also requires select RLS in general.  Also the reference docs call out you need both

But RLS, especially in storage can be confusing.
Glad you got it working.
